x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_table/snapshot_table.tsx

```diff
@@ -33,6 +33,19 @@ interface Props {
   onSnapshotDeleted: (snapshotsDeleted: Array<{ snapshot: string; repository: string }>) => void;
 }
 
+const getLastSuccessfulManagedSnapshot = (snapshots: SnapshotDetails[]) => {
```

Comment:
```suggestion
const getLastSuccessfulManagedSnapshot = (snapshots: SnapshotDetails[]): SnapshotDetails | undefined => {
```

Looks like below uses of `Boolean(lastSuccessfulManagedSnapshot)` should also become just `lastSuccessfulManagedSnapshot` so that TS can infer conditional type correctly.